lists/arraylist/arraylist.go

```diff
@@ -119,17 +117,16 @@ func (list *List[T]) IndexOf(value T) int {
 
 // Empty returns true if list does not contain any elements.
 func (list *List[T]) Empty() bool {
-	return list.size == 0
+	return len(list.elements) == 0
 }
 
 // Size returns number of elements within the list.
 func (list *List[T]) Size() int {
-	return list.size
+	return len(list.elements)
 }
 
 // Clear removes all elements from the list.
 func (list *List[T]) Clear() {
-	list.size = 0
 	list.elements = []T{}
```

Comment:
We can probably reuse the memory here instead of dropping it: `clear(list.elements[:cap(list.elements)])`

And then `list.elements = list.elements[:0]`

The reasoning behind this is that if the caller wanted to completely drop the memory, they'd simply create a new list. The semantics of Clear are that it empties the data structure but keeps the memory